app/controllers/photos_controller.rb

```diff
@@ -0,0 +1,55 @@
+class PhotosController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:show]
+
+  def index
+    @photos = current_user.photos.reverse
+  end
+
+  def new
+    @photo = Photo.new
+  end
+
+  def create
+    photo = Photo.create(photo_params.merge(user: current_user))
+    if !(photo.valid?)
+      flash[:notice] = "You must select a photo to upload"
+      redirect_to new_photo_path
+    else
+      redirect_to photos_path
+    end
+  end
+
+  def show
+    @photo = Photo.find(params[:id])
+    @likes = @photo.likes.count
```

Comment:
Little bit confusing: I'd expect `@likes` to be an array of instances of `Like`, rather than an integer.

Better to just call `@photo.likes.count` in the view